File: scripts/Interactive_Batch_Averaging2.c
#include <stdio.h>
/// @brief 
/// @param argc 
/// @param argv 
/// @return 
int main(int argc, char **argv){
int count, total, number; float average;
FILE *infile = fopen("input.dat","r");
FILE *outfile = fopen("output1.dat","w");
for (count = total = 0; fscanf(infile,"%d",&number) != EOF;){
count++; total += number;
average = (float)total/(float)count;
fprintf(outfile,"%d %f\n",count,average);
}
printf("The average is: %f\n", average);
fclose(infile);
fclose(outfile);
return 1;
}

File: scripts/Interactive_Batch_Averaging2.py
total = count = 0.0
infile = open("input.dat","r")
outfile = open("output1.dat","w")
for line in infile:
    try:
        number = int(line)
        total += number
        count += 1
        outfile.write("%d %f\n" % (count,total/count))
    except ValueError:
        break
print("The average is %f\n" % (total/count))
infile.close()
outfile.close()